source/uwp/Renderer/lib/AdaptiveShowCardActionRenderer.cpp

```diff
@@ -39,4 +41,69 @@ namespace AdaptiveNamespace
             jsonObject, elementParserRegistration, actionParserRegistration, adaptiveWarnings, element);
     }
     CATCH_RETURN;
+
+    HRESULT AdaptiveShowCardActionRenderer::BuildShowCard(_In_ IAdaptiveCard* showCard,
+                                                          _In_ IAdaptiveRenderContext* renderContext,
+                                                          _In_ IAdaptiveRenderArgs* renderArgs,
+                                                          bool isBottomActionBar,
+                                                          _Outptr_ IUIElement** uiShowCard) noexcept try
+    {
+        ComPtr<IAdaptiveHostConfig> hostConfig;
+        RETURN_IF_FAILED(renderContext->get_HostConfig(&hostConfig));
+
+        ComPtr<IAdaptiveActionsConfig> actionsConfig;
+        RETURN_IF_FAILED(hostConfig->get_Actions(&actionsConfig));
+
+        ComPtr<IAdaptiveShowCardActionConfig> showCardActionConfig;
+        RETURN_IF_FAILED(actionsConfig->get_ShowCard(&showCardActionConfig));
+
+        ABI::AdaptiveNamespace::ContainerStyle showCardConfigStyle;
+        RETURN_IF_FAILED(showCardActionConfig->get_Style(&showCardConfigStyle));
+
+        boolean wasInShowCard;
+        RETURN_IF_FAILED(renderArgs->get_IsInShowCard(&wasInShowCard));
+        RETURN_IF_FAILED(renderArgs->put_IsInShowCard(true));
+
+        ComPtr<IFrameworkElement> localUiShowCard;
+        RETURN_IF_FAILED(XamlBuilder::BuildXamlTreeFromAdaptiveCard(showCard, localUiShowCard.GetAddressOf(), renderContext, nullptr, showCardConfigStyle));
+
+        RETURN_IF_FAILED(renderArgs->put_IsInShowCard(wasInShowCard));
+
+        ComPtr<IGrid2> showCardGrid;
+        RETURN_IF_FAILED(localUiShowCard.As(&showCardGrid));
+
+        // Set the padding
+        ComPtr<IAdaptiveSpacingConfig> spacingConfig;
+        RETURN_IF_FAILED(hostConfig->get_Spacing(&spacingConfig));
+
+        UINT32 padding;
+        RETURN_IF_FAILED(spacingConfig->get_Padding(&padding));
+
+        ABI::AdaptiveNamespace::ActionMode showCardActionMode;
+        RETURN_IF_FAILED(showCardActionConfig->get_ActionMode(&showCardActionMode));
+
+        // Set the top margin
+        ComPtr<IFrameworkElement> showCardFrameworkElement;
+        RETURN_IF_FAILED(localUiShowCard.As(&showCardFrameworkElement));
+
+        UINT32 inlineTopMargin;
+        RETURN_IF_FAILED(showCardActionConfig->get_InlineTopMargin(&inlineTopMargin));
+
+        double sideMargin = (double)padding * -1;
```

Comment:
>(double)padding * -1; [](start = 28, length = 21)

nit: this could be changed for (double)(-padding)

but it would be better to use static_cast<double>(-padding)

Reply:
`padding` is unsigned, so it has to be `-static_cast<double>(padding)`, but your point stands :)

---
In reply to: [326399165](https://github.com/microsoft/AdaptiveCards/pull/3457#discussion_r326399165) [](ancestors = 326399165)